webapp/src/ts/services/form.service.ts

```diff
@@ -142,33 +142,33 @@ export class FormService {
       .then(([reports, lineage]) => this.contactSummaryService.get(contact, reports, lineage));
   }
 
-  private canAccessForm(formDoc, user, instanceData, contactSummary) {
+  private canAccessForm(formContext: EnketoFormContext) {
     return this.xmlFormsService.canAccessForm(
-      formDoc,
-      user,
-      { doc: instanceData?.contact, contactSummary: contactSummary?.context },
+      formContext.formDoc,
+      formContext.userContact,
+      {
+        doc: typeof formContext.instanceData !== 'string' && formContext.instanceData?.contact,
+        contactSummary: formContext.contactSummary?.context,
+        shouldEvaluateExpression: formContext.shouldEvaluateExpression(),
+      },
     );
   }
```

Comment:
At this point it seems like we should make the `xmlFormsService.canAccessForm` function take an `EnketoFormContext` as the parameter and then encapsulate the rest of this in that function. That seems more straightforward than doing this interim data translation here....

Reply:
I think the reason for not passing form context in the first place is because we call this function when we filter forms to be displayed for users, in the FAB menu for example or on the reports page. It will not be a form context in that case, and I don't think we should build it as form context because of that. 

https://github.com/medic/cht-core/blob/69601737292037610b1c93f27a6e454281a02cd4/webapp/src/ts/services/xml-forms.service.ts#L239